src/nvme/fabrics.c

```diff
@@ -903,8 +931,16 @@ int nvmf_get_discovery_log(nvme_ctrl_t c, struct nvmf_discovery_log **logp,
 		.rae = false,
 		.ot = false,
 	};
-	*logp = nvme_discovery_log(c, &args, max_retries);
-	return *logp ? 0 : -1;
+
+	log = nvme_discovery_log(c, &args, max_retries);
+	if (!log)
+		return -1;
+
+	for (int i = 0; i < le64_to_cpu(log->numrec); i++)
```

Comment:
We may need to move this inside `nvme_discovery_log()` so that it also applies to  `nvmf_get_discovery_wargs()`.

Reply:
Blech, I did read that `nvmf_get_discovery_wargs()` is actually calling `nvmf_get_discovery_log()`... let me fix this.

Reply:
I decided against moving this to `nvme_discovery_log()` because I wanted to avoid cluttering the 'get' function with this.

Testing took a bit longer because the software target of Linux is kind of funky when it comes to persistent discovery controllers :)